packages/core-utils/src/route.ts

```diff
@@ -425,3 +426,39 @@ export function makeRouteComparator(
     makeStringValueComparator(obj => obj.longName)
   );
 }
+
+/**
+ * Tests if a pair of colors is readable. If it is, that readble color is returned.
+ * If it is not, a more appropriate alternative is returned.
+ *
+ * Uses algorithm based on combined luminance. Values have been derived from
+ * looking at real agency color pairings. These pairings are difficult to
+ * generate for, as some colors see both white and black used by different agencies.
+ *
+ * This method therefore can accept black and white for the same background color.
```

Comment:
How about any other color?

Reply:
These aren't common, but I've updated the comment anyway